src/handler.rs

```diff
@@ -161,40 +187,25 @@ impl RequestHandler {
             }
         }
 
-        // Health endpoint logs
-        if health_request {
-            tracing::debug!(
-                "incoming request: method={} uri={}{}",
-                method,
-                uri,
-                remote_addr_str,
-            );
-        } else {
+        async move {
+            if let Some(result) = health::pre_process(&self.opts, req, &remote_addr_str) {
+                return result;
+            }
+
+            let method = req.method();
+            let headers = req.headers();
+            let uri = req.uri();
+
+            let mut uri_path = uri.path().to_owned();
+            let uri_query = uri.query();
+
+            // Health requests aren't logged here but in health module.
             tracing::info!(
                 "incoming request: method={} uri={}{}",
                 method,
                 uri,
                 remote_addr_str,
             );
```

Comment:
We want to log only when the `health` feature is disabled. Can we add a conditional over here?

Reply:
You're right. I just missed the return.